command/server.go

```diff
@@ -1686,23 +1643,15 @@ CLUSTER_SYNTHESIS_COMPLETE:
 	// Initialize the HTTP servers
 	for _, ln := range lns {
 		handler := vaulthttp.Handler(&vault.HandlerProperties{
-			Core:                         core,
-			MaxRequestSize:               ln.maxRequestSize,
-			MaxRequestDuration:           ln.maxRequestDuration,
-			DisablePrintableCheck:        config.DisablePrintableCheck,
-			UnauthenticatedMetricsAccess: ln.unauthenticatedMetricsAccess,
-			RecoveryMode:                 c.flagRecovery,
+			Core:                  core,
+			ListenerConfig:        ln.Config,
+			DisablePrintableCheck: config.DisablePrintableCheck,
+			RecoveryMode:          c.flagRecovery,
 		})
 
```

Comment:
Likely okay as-is since we built out the `lns` right before this which includes a non-nil `ln.Config` for each of the listeners, but it wouldn't hurt to nil check in here before accessing the config values.

Reply:
Done.